sdk/program/src/alt_bn128/mod.rs

```diff
@@ -72,13 +74,14 @@ impl From<u64> for AltBn128Error {
 
 impl From<AltBn128Error> for u64 {
     fn from(v: AltBn128Error) -> u64 {
+        // note: should never return 0, as it risks to be confused with syscall success
         match v {
             AltBn128Error::InvalidInputData => 1,
             AltBn128Error::GroupError => 2,
             AltBn128Error::SliceOutOfBounds => 3,
             AltBn128Error::TryIntoVecError(_) => 4,
             AltBn128Error::ProjectiveToG1Failed => 5,
-            AltBn128Error::UnexpectedError => 0,
+            AltBn128Error::UnexpectedError => 6,
```

Comment:
Is it not currently possible to cause `UnexpectedError` on bn128 operations? If so, it seems like there could be divergence between versions before the new feature gate https://github.com/anza-xyz/agave/issues/320 is activated? I am probably missing something, but just wanted to be sure.

Reply:
0 = success. That's a bug in the code.

Whether it can happen or not is... hard to tell. Add, Mul and Pairing can return Unexpected error just by looking at the code. But there's input validation happening before so... who knows!? I don't think it can happen. I'm sure we can write a program that makes it happen, it's certainly not a normal program.

Regardless, this needs to be fixed. We can't have something run, generate an error and return success.

Reply:
Okay, looking at the code, it seems fine to me.